gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -759,6 +758,19 @@ private void cancelCurrentRequest() {
     contentChannelEndOffset = -1;
   }
 
+  private void drainIterator() {
+    if (resIterator == null) {
```

Comment:
nit: maybe add the entire logic in 
```
if (resIterator != null) {
 ...
}
```

Reply:
This is a common pattern `guard-clause`. I think it is clear to avoid nesting